vendor/github.com/docker/distribution/registry/storage/linkedblobstore.go

```diff
@@ -289,11 +281,19 @@ func (lbs *linkedBlobStore) Enumerate(ctx context.Context, ingestor func(digest.
 	return nil
 }
 
-func (lbs *linkedBlobStore) mount(ctx context.Context, sourceRepo reference.Named, dgst digest.Digest) (distribution.Descriptor, error) {
+func (lbs *linkedBlobStore) mount(ctx context.Context, repoOverride distribution.RepositoryOverride, sourceRepo reference.Named, dgst digest.Digest) (distribution.Descriptor, error) {
 	repo, err := lbs.registry.Repository(ctx, sourceRepo)
 	if err != nil {
 		return distribution.Descriptor{}, err
 	}
+
+	if repoOverride != nil {
+		repo, err = repoOverride(ctx, repo)
```

Comment:
> why wouldn't we just add an optional Stat *distribution.Descriptor field to the Mount struct

@liggitt good idea, I like it. I'll amend your commit a bit to stat the source repository.

My initial feeling was that upstream code should work with wrapped objects everywhere. Not just using wrapped ones on some places and their own on others. However, the less intrusive change, the higher chances to be accepted upstream.